components/core/tests/test-ir_encoding_methods.cpp

```diff
@@ -332,6 +332,51 @@ TEMPLATE_TEST_CASE("decode_next_message_general", "[ffi][decode_next_message]",
             decode_next_message<TestType>(incomplete_message_buffer, message, timestamp));
 }
 
+// The test case only tests eight_byte_encoded_variable_t. Because the timestamp
+// size is fixed, it is easier to reverse engineer the last index of logtype in
+// the ir_buffer. The test makes modification to the logtype in the ir_buf
+// to trigger a mocked IRErrorCode_Decode_Error.
+TEST_CASE("message_decode_error", "[ffi][decode_next_message]")
+{
+    vector<int8_t> ir_buf;
+    string logtype;
+
+    string placeholder_as_string{enum_to_underlying_type(VariablePlaceholder::Dictionary)};
+    string message = "Static <\text>, dictVar1, 123, 456.7 dictVar2, 987, 654.3," +
+                     placeholder_as_string + " end of static text";
+    epoch_time_ms_t reference_ts = get_next_timestamp_for_test<eight_byte_encoded_variable_t>();
+    REQUIRE(true ==
+            encode_message<eight_byte_encoded_variable_t>(reference_ts, message, logtype,
+                                                          ir_buf));
+
+    // Timestamp is encoded as tagbyte + eight_byte_encoded_variable_t
+    size_t timestamp_encoding_size = sizeof(ffi::ir_stream::cProtocol::Payload::TimestampVal) +
+                                     sizeof(eight_byte_encoded_variable_t);
+    const size_t logtype_end_index = (ir_buf.size() - 1) - timestamp_encoding_size;
```

Comment:
* The convention in C++ is for the end index to point to the item after the last item. So ideally, we wouldn't do - 1 here, but rather when you use it on line 362.
* Minor: The convention in the STL libraries ([ex](https://cplusplus.com/reference/string/string/find/)) is to use `pos` for indices rather than `index`.